flow/util/template_filters.py

```diff
@@ -109,6 +109,49 @@ def calc_tasks(operations, name, parallel=False, allow_mixed=False):
         )
 
 
+def check_memory(operations):
+    """Check whether memory is defined by the user or not.
+
+    Parameters
+    ----------
+    operations : list
+        The operations of :class:`~._JobOperation` used to calculate the maximum memory required.
+
+    Returns
+    -------
+    bool
+        Evaluates to True if memory is defined by the user, else False.
+    """
+    return any(operation.directives["memory"] for operation in operations)
+
+
+def calc_memory(operations, parallel=False):
+    """Calculate the maximum memory to reserve for submission of operations.
+
+    Parameters
+    ----------
+    operations : list
+        The operations of :class:`~._JobOperation` used to calculate the maximum memory required.
+    parallel : bool
+        If True, operations are assumed to be executed in parallel, which means
+        that the total memory requested will be the sum of all memories requested instead of the
+        maximum memory requested. (Default value = False)
```

Comment:
I thought that all instances of the period before the `(Default value = ...)` were created by Bradley during all his refactoring and performance optimization work, but perhaps some were yours as well? In any case, period after is much more commonly used in `signac` (and other projects in the `glotzerlab` organization that use Google docstrings, for instance `freud`, `rowan`, and `coxeter`). It would be nice to make this consistent everywhere... but it's probably the lowest priority task every so I'll just point it out here for future changes 😂 .